bundles/org.openhab.binding.publictransportswitzerland/src/main/java/org/openhab/binding/publictransportswitzerland/internal/stationboard/PublicTransportSwitzerlandStationboardHandler.java

```diff
@@ -0,0 +1,187 @@
+
+    private void updateChannels(JsonElement jsonObject) throws Exception {
+        JsonArray stationboard = jsonObject.getAsJsonObject().get("stationboard").getAsJsonArray();
+        createDynamicChannels(stationboard.size());
+
+        List<String> tsvRows = new ArrayList<>();
+
+        for (int i = 0; i < stationboard.size(); i++) {
+            JsonElement jsonElement = stationboard.get(i);
+
+            JsonObject departureObject = jsonElement.getAsJsonObject();
+            JsonObject stopObject = departureObject.get("stop").getAsJsonObject();
+
+            String category = departureObject.get("category").getAsString();
+            String number = departureObject.get("number").getAsString();
+            String destination = departureObject.get("to").getAsString();
+
+            JsonElement delayElement = departureObject.get("delay");
+            String delay = "";
+            if (delayElement != null) {
+                delay = delayElement.getAsString();
+            }
+
+            Long departureTime = stopObject.get("departureTimestamp").getAsLong();
+            String track = stopObject.get("platform").getAsString();
+
+            updateState(getChannelUIDForPosition(i), new StringType(formatDeparture(category, number, departureTime, destination, track, delay)));
+            tsvRows.add(String.join("\t", category, number, departureTime.toString(), destination, track, delay));
+        }
+
+        updateState(CHANNEL_TSV, new StringType(String.join("\n", tsvRows)));
+    }
+
+    private String formatDeparture(String category, String number, Long departureTimestamp, String destination, String track, @Nullable String delay) {
+        Date departureDate = new Date(departureTimestamp * 1000);
+
+        SimpleDateFormat timeFormat = new SimpleDateFormat("HH:mm");
+        String formattedDate = timeFormat.format(departureDate);
+
+        String train = number.startsWith(category) ? category : category + number;
+        String result = String.format("%s - %s %s - Pl. %s", formattedDate, train, destination, track);
+
+        if (delay != null && !delay.isEmpty()) {
+            result += String.format(" (%s' late)", delay);
+        }
+
+        return result;
+    }
+
+    private void createDynamicChannels(int numberOfChannels) {
+        List<Channel> existingChannels = getThing().getChannelsOfGroup(dynamicChannelGroupUID.getId());
+
+        ThingBuilder thingBuilder = editThing();
+
+        for (int i = existingChannels.size(); i < numberOfChannels; i++) {
+            Channel channel = ChannelBuilder
+                    .create(getChannelUIDForPosition(i), "String")
+                    .withLabel("Departure " + (i + 1))
+                    .build();
+            thingBuilder.withChannel(channel);
+        }
+
+        updateThing(thingBuilder.build());
+    }
+
+    private ChannelUID getChannelUIDForPosition(int position) {
+        return new ChannelUID(dynamicChannelGroupUID, String.valueOf(position + 1));
```

Comment:
Just a remark (not a request for change): is just a number a good id for the channel ?

Reply:
Since they are in the departures group I would argue so.